pandas/core/dtypes/missing.py

```diff
@@ -562,12 +562,29 @@ def _array_equivalent_datetimelike(left: np.ndarray, right: np.ndarray):
 
 
 def _array_equivalent_object(left: np.ndarray, right: np.ndarray, strict_nan: bool):
-    if not strict_nan:
-        # isna considers NaN and None to be equivalent.
-
-        return lib.array_equivalent_object(ensure_object(left), ensure_object(right))
-
-    for left_value, right_value in zip(left, right):
+    left = ensure_object(left)
```

Comment:
can we get here in cases that are not already object?  seems like if that happens then something has gone wrong?